repose-aggregator/components/filters/destination-router-filter/src/main/scala/org/openrepose/filters/destinationrouter/DestinationRouterFilter.scala

```diff
@@ -95,7 +91,13 @@ class DestinationRouterFilter @Inject()(configurationService: ConfigurationServi
             logger.error("The destination could not be added -- the destinations attribute was of an unknown type")
         }
 
-        mbcsRoutedReponse.mark(target.getId)
+        metricsServiceOption.foreach(metricsService =>
```

Comment:
I agree about the complexity. In this case, the target ID is part of the metric, so the metric itself would have to be fetched when the configuration is updated. That's no issue. In other cases, though, there is a dynamic component to the metric name that is only available when processing the request. I used this form everywhere to be consistent, but it's true that it isn't necessarily the best form to use in all cases.

As for the mark method without an argument, that is all the underlying library provides. At least, mark with a `String` parameter is not supported by the underlying metrics library. That is what our custom `ByCategory` metric classes were buying us. In the end we get the same metrics, it's just a matter of what the call looks like.

Let me try to clean up this class a little bit, then you can tell me what you think.